code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -418,53 +418,53 @@
 	togglelock(user)
 
 /mob/living/silicon/robot/attackby(obj/item/W, mob/user, params)
-	if(W.tool_behaviour == TOOL_WELDER && (user.a_intent != INTENT_HARM || user == src))
+	if(length(user.progressbars))
+		if(W.tool_behaviour == TOOL_WELDER || istype(W, /obj/item/stack/cable_coil))
+			user.changeNext_move(CLICK_CD_MELEE)
+			to_chat(user, "<span class='notice'>You are already busy!</span>")
+			return
+	if(W.tool_behaviour == TOOL_WELDER && (user.a_intent != INTENT_HARM))
 		user.changeNext_move(CLICK_CD_MELEE)
-		if (!getBruteLoss())
-			to_chat(user, "<span class='warning'>[src] is already in good condition!</span>")
+		if(src == user)
+			to_chat(user, "<span class='notice'>You are unable to properly repair yourself, seek assistance.</span>")
```

Comment:
"You try to reach your damaged shell with the welder, but your cyborg arms lack the versatility to repair yourself."